google/cloud/pubsub_v1/subscriber/message.py

```diff
@@ -252,6 +268,10 @@ def ack(self) -> None:
             https://cloud.google.com/pubsub/docs/exactly-once-delivery."
 
         """
+        if self.opentelemetry_data:
+            self.opentelemetry_data.add_subscribe_span_event("ack start")
+            self.opentelemetry_data.add_process_span_event("ack called")
```

Comment:
This span should be added when the user calls msg.Ack(), rather than when the internal library ack is called.

Reply:
Ok reading this, it's the subscribe span event that is actually in the wrong place. That should start with the RPC is invoked, not when the user calls msg.Ack.

Reply:
Got it. Moved the `add_subscribe_span_event("ack start")` from message.py to dispatcher.py(where the actual RPC call is made). Did this for both ack and ack_with response. Similarly for nack / nack_with_response and modify_ack_deadline and modify_ack_deadline_with_response